lib/core/router.coffee

```diff
@@ -59,8 +68,16 @@ module.exports = class KDRouter extends KDObject
   startListening:->
     return no  if @isListening # make this action idempotent
     @isListening = yes
+
+    if isWindowLoaded()
+      @blockPopstateIfWindowLoaded = no
+    else
+      @blockPopstateIfWindowLoaded = yes
+      window.addEventListener 'load', @bound 'handleWindowLoad'
```

Comment:
I think, if you only edit `::startListenining` as below it would be cleaner and enough.

```coffee
startListening: ->
  
  return no  if @isListening # make this action idempotent
  
  # to avoid safari's inconsistent initial popstate firing 
  unless document.readyState is 'complete'
    return document.onreadystatechange = ->
      @startListening()  if document.readyState is 'complete'

  @isListening = yes
  window.addEventListener 'popstate', @bound "popState"

  return yes
```